spec/diagnose_spec.rb

```diff
@@ -946,8 +947,29 @@
   end
 end
 
+RSpec.describe "Running the diagnose command with report submission error" do
+  before(:all) do
+    MockServer.auth_response_code = 200
+    MockServer.diagnose_response_code = 500
+    @runner = init_runner(:prompt => "y")
+    @runner.run
+    @received_report = MockServer.last_diagnose_report
+  end
+
+  it "prints an error about report submission" do
+    send_report = section(:send_report)
+    expect(send_report).to include(
+      "  Error: Something went wrong while submitting the report to AppSignal.",
+      "  Response code: 500",
+      "  Response body:",
+      "{\"token\":\"diag_support_token\"}"
```

Comment:
Maybe we can set another response body on failure so it doesn't look like it should have worked? To avoid confusion later on. Currently it prints the response you'd get when the report was successfully submitted.

The response body `some error` on any response code that's not `200` would probably be fine.